### DISCUSSION TITLE
CORS error when connecting to mongo atlas

### DISCUSSION THREAD

**Discussion by thendodev** [honojs/hono#2856] [Q&A] [answered]
Hey and GOD bless, first of all i want to say thank you for your help, its much appreciated, i have been stuck on this issue for a while now, just deployed to vercel and also included the ability to work with a deployed version of the db, i am using a mongo client wrapper called Papr js for models and connection, it seems like everything is fine hen i connect to the local db on my machine but as soon as i try to connect to atlas i start getting cors errors this weird because i am using next js and my hono backend is in the same repo as my front end 
the header for the request basically looks like this
```
POST
	http://localhost:3000/api/auth/login
Transferred554 B (0 B size)
Referrer Policystrict-origin-when-cross-origin
Request PriorityHighest
DNS ResolutionSystem

	
Accept
	application/json, text/plain, */*
Accept-Encoding
	gzip, deflate, br, zstd
Accept-Language
	en-US,en;q=0.5
Connection
	keep-alive
Content-Length
	54
Content-Type
	application/json
Host
	localhost:3000
Origin
	http://localhost:3000
Priority
	u=1
Referer
	http://localhost:3000/
Sec-Fetch-Dest
	empty
Sec-Fetch-Mode
	cors
Sec-Fetch-Site
	same-origin
User-Agent
	Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:127.0) Gecko/20100101 Firefox/127.0
```

and  my db connection
```
import { envServer } from '@/global/envServer';
import { MongoClient } from 'mongodb';
import Papr from 'papr';

const papr = new Papr();
export let client: MongoClient | null = null;

const getDbConnection = () => {
  switch (envServer.NEXT_PUBLIC_ENVIRONMENT) {
    case undefined:
      return console.log('no enviroment defined');
    case 'development':
      return envServer.DEV_MONGO_URI;
    case 'production':
      return envServer.PROD_MONGO_URI;
    case 'uat':
      return envServer.UAT_MONGO_URI;
    default:
      envServer.DEV_MONGO_URI;
  }
};

//

export async function Dbconnect(): Promise<MongoClient | void> {
  const dbConnection = getDbConnection();
  if (!dbConnection) return console.log('no connection');
  client = await MongoClient.connect(dbConnection);

  papr.initialize(client.db(envServer.DATABASE_NAME));

  await papr.updateSchemas();
}

export async function Dbdisconnect() {
  if (!client) return;
  await client.close();
}

export default papr;

```

**Comment by 6km** [comment]
@thendodev Please provide more information and, if possible, screenshots of the errors you're getting.

**Reply by thendodev** [reply]
whats happening is when i connect to my local db everything works fine but when i try and use the atlas hosted db, i start getting cors errors, i am not sure how this is happening or why


![Screenshot 2024-05-29 at 16 55 51 (2)](https://github.com/honojs/hono/assets/98970430/7d0b0557-7650-4899-a8ee-59e52416916e)



here is my server file as well

```
mport { OpenAPIHono } from '@hono/zod-openapi';
import { apiReference } from '@scalar/hono-api-reference';
import { swaggerUI } from '@hono/swagger-ui';
import { login } from './routes/auth/login/login';
import { signUp } from './routes/auth/sign-up/sign-up';
import { opt } from './routes/auth/otp/otp';
import { logger } from './middleware/logger';
import { recruitment } from './routes/recruitment';
const app = new OpenAPIHono();

//initailaize swagger
app.doc31('/api/swagger.json', {
  openapi: '3.1.0',
  info: {
    title: 'API Reference',
    version: '1.0.0',
  },
});

//set up swagger endpoint
app.get(
  'api/swagger',
  swaggerUI({
    url: '/api/swagger.json',
  }),
);

//set up scalar endpoint
app.get(
  '/api/scalar',
  apiReference({
    spec: {
      url: '/api/swagger.json',
    },
  }),
);

//register middleware

app.use(logger);

//authentification end points
app.route('/', login);
app.route('/', signUp);
app.route('/', opt);

//recruitment
app.route('/api/recruitment', recruitment);

export { app };
```

**Reply by 6km** [reply]
@thendodev The way Firefox handles HTTP requests is different from other browsers. It throws an NS_BINDING_ABORTED error when the load on the page is interrupted by something else, for example, page navigation.

Can you try another browser and see if it works as expected?

**Reply by thendodev** [reply]
Chrome seems to not work aswell


![Screenshot 2024-05-29 at 18 24 56 (2)](https://github.com/honojs/hono/assets/98970430/ad991b91-2e76-4dde-b92c-a1937df24757)
![Screenshot 2024-05-29 at 18 24 43 (2)](https://github.com/honojs/hono/assets/98970430/7150006b-1a5d-4f5c-aa3f-5c1857315a89)

**Reply by 6km** [reply]
@thendodev I'm investigating this issue. It looks like you're using Axios. can you show me the Axios error from the console?
By the way, are you using any CORS middleware? if yes, please show me its configuration.

**Reply by thendodev** [reply]
'timeout of 3000ms exceeded' and no, i am not using any cors middleware

**Reply by 6km** [reply, marked as answer]
@thendodev I guess there's something on the server side that delays server response. It might be a timeout, database issues, etc.
If you have set a timeout for Axios's HTTP client, try to make it longer.
Also, check the process of logging in from the server side, it might have a delay or something else.

**Reply by thendodev** [reply]
Hey, so i commented out the logger, it seems like the requests are being processed just fine, but the request it self is timeing out, might this be an issue mongo free tier just taking too long and axios canceling the request ? i think i need to investigate but for the most part, your help has been much appreciated, also, a big fan of hono, decided to try it out and i love what you guys are doing

**Reply by 6km** [reply]
@thendodev Pleased to hear that you loved Hono!

By the way, if the *`timeout of 3000ms exceeded`* error is by the Axios client, increase the timeout. Otherwise, the issue might be due to the limitations of Mongo's free tier.
